### PR TITLE
Enhance Lighthouse Integration - Multiple URLs, Form Factor Configuration, and New Metrics

### PR DESCRIPTION
**Pull request by baerbradford**
### What does this PR do?

This PR makes three modifications to the Lighthouse integration.

1. Enhances the integration to allow for configuring multiple URLs to score. Lighthouse is run in a loop against each configured URL. A new "urls" configuration value is read in the conf.yaml. For backward compatibility with older files, the "url" param is still processed if present.
2. Adds a form_factor param as a configurable value. This value is just passed through to Lighthouse when it is run.
3. Adds new performance metrics reported by lighthouse.

### Motivation

I wanted to sample multiple pages in my application for SEO performance and report these values in Datadog. In the current implementation only allowing a single URL to be configured, I wouldn't be able to do this without multiple agents. Lighthouse scoring should be performed in isolation to reduce variance, so multiple agents on a single machine wouldn't work. Running multiple agents on multiple machines would scale costs too much. So modifying the integration to run against multiple URLs sequentially seemed the best solution.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: lighthouse/datadog_checks/lighthouse/data/conf.yaml.example
Comment (yzhan289) [inline lighthouse/datadog_checks/lighthouse/data/conf.yaml.example:15]: Change for validation:
```suggestion
    ## @param url - string - optional
    ## url of the website to get lighthouse metrics of. This option is deprecated.
    ## ex: "https://www.datadoghq.com"
    #
    url: <URL_PLACEHOLDER>
```

**Review by yzhan289 — CHANGES_REQUESTED** [review]
Thanks for the PR. I see you are directly modifying `conf.yaml.example`. We recommend having a `spec.yaml` file in the `assets` directory of the check, and running `ddev validate config lighthouse -s` so it autogenerates the `conf.yaml.example` file. More info can be found here: https://datadoghq.dev/integrations-core/meta/config-specs/ and https://docs.datadoghq.com/developers/integrations/new_check_howto/?tab=configurationtemplate#create-the-check-assets

**Comment by hithwen** [comment]
Pinging @ericmustin as integration maintainer

**Review by ericmustin — APPROVED** [review]
Thanks for the ping @hithwen . hope all is well! 

Broadly speaking , it looks like the changes requested from @yzhan289 have been implemented, and this PR looks good to me @baerbradford and I think significantly improves this integration. Awesome stuff!

Also worth noting that I am no longer am an employee of Datadog (still a happy user though! 😄 ). I'm not as up to date with agent-integration best practices as I probably ought to be. If there's anyone who'd like to take over as maintainer of this integration, it would probably make sense, as in my current role I'm not going to be able to stay as up-to-date as I should be with the intricacies of best practice for custom/"extra" Agent integrations, and give the Lighthouse integration the love it deserves. 🙃 

Thanks again for this PR!

**Comment by ericmustin** [comment]
err, actually it looks like theres still a missing `spec.yaml` file for the auto-generated stuff to run, as @yzhan289 pointed out, but the actual code LGTM so i'm gonna leave my approval on this and defer to the current employees on the formatting bit.

**Comment by baerbradford** [comment]
Yeah... I gave the spec.yaml approach a shot, but had a hard time with it. I wasn't easily able to make it generate a comparable output. I noticed many of the integrations didn't have one, though, so I just took the change to make the example config file valid.

If generating the example config from a spec is a requirement, I may need someone more familiar with these to get it working.

**Comment by hithwen** [comment]
Hallo, @baerbradford. Do not worry about the config specs. Migrating to that model is an effort outside of the scope of this PR and it is only a strict requirement for new integrations.

**Comment by hithwen** [comment]
> Also worth noting that I am no longer am an employee of Datadog (still a happy user though! 😄 ). I'm not as up to date with agent-integration best practices as I probably ought to be. If there's anyone who'd like to take over as maintainer of this integration, it would probably make sense, as in my current role I'm not going to be able to stay as up-to-date as I should be with the intricacies of best practice for custom/"extra" Agent integrations, and give the Lighthouse integration the love it deserves. 🙃
> 
> Thanks again for this PR!

Well, the majority of extras integrations are not maintained by anyone at Datadog. The definition of this repo is for community integrations not officially maintained by Datadog (vs core integrations). The role of the maintainer is to address PRs and potential issues with the integration.

File: lighthouse/datadog_checks/lighthouse/data/conf.yaml.example
Comment (sarah-witt) [inline lighthouse/datadog_checks/lighthouse/data/conf.yaml.example:15]:
```suggestion
    ## @param url - string - optional
    ## url of the website to get lighthouse metrics of. This option is deprecated.
    ## ex: "https://www.datadoghq.com"
    #
    # url: <URL_PLACEHOLDER>
```
Let's comment this out since it is now deprecated

File: lighthouse/datadog_checks/lighthouse/data/conf.yaml.example
Comment (sarah-witt) [inline lighthouse/datadog_checks/lighthouse/data/conf.yaml.example:24]:
```suggestion
      - <URL_PLACEHOLDER1>
      - <URL_PLACEHOLDER2>
```

File: lighthouse/metadata.csv
Comment (sarah-witt) [inline lighthouse/metadata.csv:21]: Could we make these descriptions more descriptive, for example:
```suggestion
lighthouse.speed_index,gauge,,millisecond,,The number of milliseconds it takes for the contents of a page to be visibly populated.,-1,lighthouse,speed_index
lighthouse.dom_size,gauge,,,,The current size of the DOM. A large DOM will increase memory usage cause longer style calculations and produce costly layout reflows.,-1,lighthouse,dom_size
```

File: lighthouse/datadog_checks/lighthouse/data/conf.yaml.example
Comment (sarah-witt) [inline lighthouse/datadog_checks/lighthouse/data/conf.yaml.example:21]: Have you tried creating multiple check instances, each with one url? This should allow you to check multiple urls with one agent, unless there is a specific limitation with the check. It is  a smoother user experience to provide multiple urls in one instance, though.

File: lighthouse/datadog_checks/lighthouse/lighthouse.py
Comment (sarah-witt) [inline lighthouse/datadog_checks/lighthouse/lighthouse.py:18]: Can we alert the user  if `form_factor` is not `mobile` or `desktop`, and set `form_factor` to `None` if so?

File: lighthouse/tests/test_lighthouse.py
Comment (sarah-witt) [inline lighthouse/tests/test_lighthouse.py:34]: It would be nice to have a test using `form_factor` as well as a test using the new `urls` config option.

File: lighthouse/datadog_checks/lighthouse/data/conf.yaml.example
Comment (sarah-witt) [inline lighthouse/datadog_checks/lighthouse/data/conf.yaml.example:36]:
```suggestion
    # form_factor: <FORM_FACTOR_PLACEHOLDER>
```
Let's comment this out by default since it's optional

**Review by sarah-witt — COMMENTED** [review]
Hi @baerbradford, thanks a lot for the PR! I left some comments on integration style and formatting

**Comment by fanny-jiang** [comment]
Hi @baerbradford, have you had a chance to address Sarah's comments?

**Comment by durgell-bl** [comment]
@fanny-jiang @sarah-witt 

As it seems that the original contributor seems not to be responding, is it possible from DataDog's side to allocate some resources in order to finish this integration? As this PR improves this integration will be welcome for any of us out there using it.

**Comment by yzhan289** [comment]
Since the contributor hasn't been responding, I'll take over implementing this PR. However, I'll keep @baerbradford as the maintainer for now unless there's a new maintainer that can take this integration.

File: lighthouse/datadog_checks/lighthouse/data/conf.yaml.example
Comment (yzhan289) [reply lighthouse/datadog_checks/lighthouse/data/conf.yaml.example:21]: This can be a future improvement. I'll try to keep as much of the contributor's content the same for this PR since the maintainer had approved this already.

**Review by yzhan289 — APPROVED** [review]

**Pull request merged**
